plugins/experimental/slice/prefetch.cc

```diff
@@ -0,0 +1,134 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+/**
+ * @file prefetch.cpp
+ * @brief Background fetch related classes (header file).
+ */
+
+#include "ts/ts.h" /* ATS API */
+#include "prefetch.h"
+
+bool
+BgBlockFetch::schedule(Data *const data, int blocknum)
+{
+  bool ret         = false;
+  BgBlockFetch *bg = new BgBlockFetch(blocknum);
+  if (bg->fetch(data)) {
```

Comment:
This might be a bit better

```
BgBlockFetch::schedule(Data *const data, int blocknum)
{
  bool ret         = false;
  std::unique_ptr<BgBlockFetch> bg = std::make_unique<BgBlockFetch>(blocknum);
  if (bg->fetch(data)) {
    ret = true;
  }
  return ret;
}
```

Reply:
I have an explicit deleter for BgBlockFetch to also close the continuation when the fetching is completed